apps/enc.c

```diff
@@ -512,18 +574,45 @@ int enc_main(int argc, char **argv)
             goto end;
         }
 
+        if ((EVP_CIPHER_flags(cipher) & EVP_CIPH_FLAG_AEAD_CIPHER)) {
+            if (!EVP_CipherUpdate(ctx, NULL, &outlen, (unsigned char*)aad, strlen(aad))) {
```

Comment:
Shouldn't we include the magic/salt etc into the AAD? i.e. everything in the file except the tag itself should be covered somehow by the integrity guarantee.